File: hiroo-py/hiroo/intents.py
from enum import IntFlag


class Intents(IntFlag):
    GUILDS                   = 1 << 0
    GUILD_MEMBERS            = 1 << 1   # privileged
    GUILD_BANS               = 1 << 2
    GUILD_VOICE_STATES       = 1 << 3
    GUILD_PRESENCES          = 1 << 4   # privileged
    GUILD_MESSAGES           = 1 << 5
    GUILD_MESSAGE_REACTIONS  = 1 << 6
    GUILD_MESSAGE_TYPING     = 1 << 7
    DIRECT_MESSAGES          = 1 << 8
    DIRECT_MESSAGE_REACTIONS = 1 << 9
    DIRECT_MESSAGE_TYPING    = 1 << 10
    MESSAGE_CONTENT          = 1 << 15  # privileged
    GUILD_SCHEDULED_EVENTS   = 1 << 16

    @classmethod
    def none(cls) -> "Intents":
        return cls(0)

    @classmethod
    def default(cls) -> "Intents":
        """Everything that's free — without privileged intents."""
        return cls(
            cls.GUILDS | cls.GUILD_BANS | cls.GUILD_VOICE_STATES |
            cls.GUILD_MESSAGES | cls.GUILD_MESSAGE_REACTIONS |
            cls.GUILD_MESSAGE_TYPING | cls.DIRECT_MESSAGES |
            cls.DIRECT_MESSAGE_REACTIONS | cls.DIRECT_MESSAGE_TYPING |
            cls.GUILD_SCHEDULED_EVENTS
        )

    @classmethod
    def all(cls) -> "Intents":
        out = cls(0)
        for m in cls:
            out |= m
        return out

File: hiroo-java/src/main/java/tech/intave/hiroo/Intents.java
package tech.intave.hiroo;

/**
 * Gateway intents. Combine with bitwise OR. Privileged intents
 * (GUILD_MEMBERS, GUILD_PRESENCES, MESSAGE_CONTENT) must be enabled
 * in the bot dashboard before the gateway will accept them.
 */
public final class Intents {
    public static final int GUILDS                   = 1 << 0;
    public static final int GUILD_MEMBERS            = 1 << 1;   // privileged
    public static final int GUILD_BANS               = 1 << 2;
    public static final int GUILD_VOICE_STATES       = 1 << 3;
    public static final int GUILD_PRESENCES          = 1 << 4;   // privileged
    public static final int GUILD_MESSAGES           = 1 << 5;
    public static final int GUILD_MESSAGE_REACTIONS  = 1 << 6;
    public static final int GUILD_MESSAGE_TYPING     = 1 << 7;
    public static final int DIRECT_MESSAGES          = 1 << 8;
    public static final int DIRECT_MESSAGE_REACTIONS = 1 << 9;
    public static final int DIRECT_MESSAGE_TYPING    = 1 << 10;
    public static final int MESSAGE_CONTENT          = 1 << 15;  // privileged
    public static final int GUILD_SCHEDULED_EVENTS   = 1 << 16;

    private Intents() {}

    /** Everything that's free — no privileged intents. */
    public static int defaults() {
        return GUILDS | GUILD_BANS | GUILD_VOICE_STATES
             | GUILD_MESSAGES | GUILD_MESSAGE_REACTIONS | GUILD_MESSAGE_TYPING
             | DIRECT_MESSAGES | DIRECT_MESSAGE_REACTIONS | DIRECT_MESSAGE_TYPING
             | GUILD_SCHEDULED_EVENTS;
    }

    public static int all() {
        return GUILDS | GUILD_MEMBERS | GUILD_BANS | GUILD_VOICE_STATES
             | GUILD_PRESENCES | GUILD_MESSAGES | GUILD_MESSAGE_REACTIONS
             | GUILD_MESSAGE_TYPING | DIRECT_MESSAGES | DIRECT_MESSAGE_REACTIONS
             | DIRECT_MESSAGE_TYPING | MESSAGE_CONTENT | GUILD_SCHEDULED_EVENTS;
    }
}
